tests/kas_messaging.spec.ts

```diff
@@ -104,3 +117,75 @@ test('Browse messages', async ({ page }) => {
   await expect(messageDetail.locator('dt:has-text("Offset")')).toHaveCount(1);
   await expect(messageDetail.locator('dd:has-text("key-")')).toHaveCount(1);
 });
+
+const filters = [FilterGroup.offset, FilterGroup.timestamp, FilterGroup.epoch, FilterGroup.latest];
+for (const filter of filters) {
+  test(`Filter messages by ${filter}`, async ({ page }) => {
+    // Today and tomorrow date
+    const today = new Date();
+    const tomorrow = new Date();
+    tomorrow.setDate(today.getDate() + 1);
+
+    await navigateToMessages(page, testInstanceName, testTopicName);
+
+    await refreshMessages(page);
+
+    switch (filter) {
+      case FilterGroup.offset: {
+        await pickFilterOption(page, FilterGroup.offset);
+        await filterMessagesByOffset(page, '0', '20', Limit.ten);
+
+        let messageTable = await page.locator('table[aria-label="Messages table"] >> tbody >> tr');
```

Comment:
```suggestion
        const messageTable = await page.locator('table[aria-label="Messages table"] >> tbody >> tr');
```

Reply:
Cannot be constant, see L147+

Reply:
removing the `await` will make the locator be re-evaluated when used, maybe is better in this case ...

Reply:
I change it and it works fine